osbuild/util/containers.py

```diff
@@ -127,4 +148,32 @@ def container_source(image):
             os.symlink(image_filepath, tmp_source)
 
         image_source = f"{container_format}:{tmp_source}"
-        yield image_name, image_source
+        yield image_source
+
+
+@contextmanager
+def container_source(image):
+    image_filepath = image["filepath"]
+    container_format = image["data"]["format"]
+    image_name = image["data"]["name"]
+
+    if container_format not in ("dir", "oci-archive", "containers-storage"):
+        raise RuntimeError(f"Unknown container format {container_format}")
+
+    if container_format == "containers-storage":
+        with containers_storage_source(image, image_name, image_filepath, container_format) as image_source:
+            yield image_name, image_source
+    else:
+        with dir_oci_archive_source(image, image_name, image_filepath, container_format) as image_source:
+            yield image_name, image_source
+
+
+def get_host_storage(storage_conf):
+    """
+    Read the host storage configuration.
+    """
+    import toml  # importing this at the top level will break the buildroot
```

Comment:
Thanks for the comment

Reply:
Is this because it's called by the source itself or because it's required to be in the buildroot to work?
If we need it for a source, then it should become a dependency of osbuild.  If it'll run in the buildroot when needed, then it should be installed when necessary.

Or am I misunderstanding?

Reply:
So I extracted this function to `util/containers.py` because it's used in the inputs which needs `toml` in the buildroot. The other caller is in the source which needs `toml` on the host. 

It's only really `containers-storage` input that needs this, for example, this function would never be called for containers coming from `skopeo`. So it seemed a bit heavy handed to make it a requirement for images that don't really need it. Does that clear it up and would we prefer to make it a requirement instead?